### PR TITLE
Overriding default NTP settings (bsc#1160080)

### PR DESCRIPTION
**Pull request by tbazant**
Deployment section about time synchronization was misleading, this update fixes it and describes how to override the default DeepSea's configuration.

### CODE REVIEW

File: xml/admin_ceph_troubleshooting.xml [hunk 1]
```diff
@@ -289,11 +289,49 @@ ID WEIGHT TYPE NAME    UP/DOWN REWEIGHT PRIMARY-AFFINITY
...
 
   <para>
-   Time synchronization is managed with NTP (see
-   <link xlink:href="http://en.wikipedia.org/wiki/Network_Time_Protocol"/>).
-   Set each node to synchronize its time with one or more NTP servers,
-   preferably to the same group of NTP servers. If the time skew still occurs
-   on a node, follow these steps to fix it:
+   By default, &deepsea; uses the &adm; as the time server for other
+   cluster nodes. Therefore, if the &adm; is not virtualized, select one or
+   more time servers/pools, and synchronize its local time against them.
```
Comment (asettle) [inline xml/admin_ceph_troubleshooting.xml]:
```suggestion
   more time servers or pools, and synchronize the local time against them.

File: xml/admin_install_salt.xml [hunk 2]
```diff
@@ -475,21 +475,47 @@
...
    <step>
     <para>
-     Select one or more time servers/pools, and synchronize the local time
-     against them. Verify that the time synchronization service is enabled on
-     each system start-up. You can use the <command>yast ntp-client</command>
-     command found in a <package>yast2-ntp-client</package> package to
-     configure time synchronization.
+     By default, &deepsea; will use the &adm; as the time server for other
+     cluster nodes. Therefore, if the &adm; is not virtualized, select one or
+     more time servers/pools, and synchronize its local time against them.
```
Comment (asettle) [inline xml/admin_install_salt.xml]:
```suggestion
     more time servers or pools, and synchronize the local time against them.

File: xml/admin_install_salt.xml [hunk 3]
```diff
@@ -475,21 +475,47 @@
    </step>
    <step>
     <para>
-     Select one or more time servers/pools, and synchronize the local time
-     against them. Verify that the time synchronization service is enabled on
-     each system start-up. You can use the <command>yast ntp-client</command>
-     command found in a <package>yast2-ntp-client</package> package to
-     configure time synchronization.
+     By default, &deepsea; will use the &adm; as the time server for other
```
Comment (asettle) [inline xml/admin_install_salt.xml]:
```suggestion
     By default, &deepsea; uses the &adm; as the time server for other

File: xml/admin_install_salt.xml [hunk 4]
```diff
@@ -475,21 +475,47 @@
...
     </para>
-    <tip>
-     <para>
-      Virtual machines are not reliable NTP sources.
-     </para>
-    </tip>
     <para>
-     Find more information on setting up NTP in
-     <link xlink:href="https://www.suse.com/documentation/sles-15/book_sle_admin/data/sec_ntp_yast.html"/>.
+     If the &adm; is a virtual machine, you need to provide better time sources
```
Comment (asettle) [inline xml/admin_install_salt.xml]:
```suggestion
     If the &adm; is a virtual machine, provide better time sources

File: xml/admin_ceph_troubleshooting.xml [hunk 5]
```diff
@@ -289,11 +289,49 @@ ID WEIGHT TYPE NAME    UP/DOWN REWEIGHT PRIMARY-AFFINITY
...
+   Verify that the time synchronization service is enabled on each system
+   start-up. Find more information on setting up time synchronization in
+   <link
+    xlink:href="https://www.suse.com/documentation/sles-15/book_sle_admin/data/sec_ntp_yast.html"/>.
+  </para>
+  <para>
+   If the &adm; is a virtual machine, you need to provide better time sources
```
Comment (asettle) [inline xml/admin_ceph_troubleshooting.xml]:
```suggestion
   If the &adm; is a virtual machine, provide better time sources

**Comment by swiftgist** [comment]
Just to verify, this part of the manual is somewhat of an appendix or set of examples explaining exceptions.  Does the reference to this section mention `salt '*' state.apply ceph.time`?  

Otherwise, it sounds good.

**Comment by tbazant** [comment]
> Just to verify, this part of the manual is somewhat of an appendix or set of examples explaining exceptions. Does the reference to this section mention `salt '*' state.apply ceph.time`?
> 
> Otherwise, it sounds good.
No, `salt '*' state.apply ceph.time` is only mentioned in a section on how to disable time synchronization. Should it be run after `salt '*' saltutil.pillar_refresh` in all these cases?

**Comment by swiftgist** [comment]
> > Just to verify, this part of the manual is somewhat of an appendix or set of examples explaining exceptions. Does the reference to this section mention `salt '*' state.apply ceph.time`?
> > Otherwise, it sounds good.
> No, `salt '*' state.apply ceph.time` is only mentioned in a section on how to disable time synchronization. Should it be run after `salt '*' saltutil.pillar_refresh` in all these cases?

Stage 0 or calling the above command directly are what actually apply the configuration.  If the reference from the other section of the manual that instructs to run Stage 0 has this reference on how to customize the time servers, then it's fine.  If this is more like a standalone appendix, then technically we just explained how to change a configuration file and did nothing with it.

**Pull request merged**